easybuild/tools/docs.py

```diff
@@ -160,3 +147,115 @@ def avail_easyconfig_params(easyblock, output_format):
         FORMAT_TXT: avail_easyconfig_params_txt,
     }
     return avail_easyconfig_params_functions[output_format](title, grouped_params)
+
+def generic_easyblocks():
+    """
+    Compose overview of all generic easyblocks
+    """
+    modules = import_available_modules('easybuild.easyblocks.generic')
+    docs = []
+    seen = []
+
+    for m in modules:
+        for name,obj in inspect.getmembers(m, inspect.isclass):
+            eb_class = getattr(m, name)
+            # skip imported classes that are not easyblocks
+            if eb_class.__module__.startswith('easybuild.easyblocks.generic') and name not in seen:
+                docs.append(doc_easyblock(eb_class))
+                seen.append(name)
+
+    toc = ['.. contents:: Available generic easyblocks', '    :depth: 1', '']
+
+    return toc + sorted(docs)
+
+
+def doc_easyblock(eb_class):
+    """
+    Compose overview of one easyblock given class object of the easyblock in rst format
+    """
+    classname = eb_class.__name__
+
+    common_params = {
+        'ConfigureMake' : ['configopts', 'buildopts', 'installopts'],
+        # to be continued
+    }
+
+    lines = [
+        '``' + classname + '``',
+        '=' * (len(classname)+4),
+        '',
+    ]
+
+    bases = ['``' + base.__name__ + '``' for base in eb_class.__bases__]
+    derived = '(derives from ' + ', '.join(bases) + ')'
+    lines.extend([derived, ''])
+
+    # Description (docstring)
+    lines.extend([eb_class.__doc__.strip(), ''])
+
+    # Add extra options, if any
+    if eb_class.extra_options():
+        extra_parameters = 'Extra easyconfig parameters specific to ' + backtick(classname) + ' easyblock'
+        lines.extend([extra_parameters, '-' * len(extra_parameters), ''])
+        ex_opt = eb_class.extra_options()
+
+        titles = ['easyconfig parameter', 'description', 'default value']
+        values = [[backtick(key) for key in ex_opt], [val[1] for val in ex_opt.values()], [backtick(str(quote_str(val[0]))) for val in ex_opt.values()]]
+
+        lines.extend(mk_rst_table(titles, values))
+        lines.append('')
+
+    if classname in common_params:
+        commonly_used = 'Commonly used easyconfig parameters with ' + backtick(classname) + ' easyblock'
+        lines.extend([commonly_used, '-' * len(commonly_used)])
+
+        for opt in common_params[classname]:
+            param = '* ' + backtick(opt) + ' - ' + DEFAULT_CONFIG[opt][1]
+            lines.append(param)
+
+
+    if classname + '.eb' in os.listdir(os.path.join(os.path.dirname(__file__), 'doc_examples')):
```

Comment:
these need to come from somewhere else, i.e. a location specified as an argument to `doc_easyblock` (which may need another name, although I can't think of one in 1-2-3)

also, don't include example .eb files here; at best, you can include some in the tests (next to the test easyconfigs we have already), so you can add a unit test for this code